packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -48,6 +57,217 @@ export async function fetchSnapshot(
             eventName: "fetchSnapshot",
             headers: Object.keys(headers).length !== 0 ? true : undefined,
         },
-        async () => epochTracker.fetchAndParseAsJSON<IOdspSnapshot>(url, { headers }, "snapshotTree"),
+        async () => snapshotUploader(url, { headers }),
     );
 }
+
+export async function fetchLatestSnapshotCore(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    tokenFetchOptions: TokenFetchOptions,
+    snapshotOptions: ISnapshotOptions | undefined,
+    logger: ITelemetryLogger,
+    snapshotUploader: (url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<IOdspSnapshot>>,
+    persistedCache: IPersistedCache,
+): Promise<{
+    odspSnapshot: ISnapshotCacheValue,
+    cacheP?: Promise<void>,
```

Comment:
This seems a bit weird (promise inside of promise).
Maybe re-enrage code (if it's not too hard) such that we return {promise<ISnapshotCacheValue>, promise<void>} ?

Reply:
But it is an async function. So whatever we return should be promise.

Reply:
Yes it works fine, just looks a bit weird. :)